app/lib/actions/pulp3/repository/update_remote.rb

```diff
@@ -1,14 +1,15 @@
 module Actions
   module Pulp3
     module Repository
-      class UpdateRemote < Pulp3::Abstract
+      class UpdateRemote < Pulp3::AbstractAsyncTask
         def plan(repository, smart_proxy)
+          repository.backend_service(smart_proxy).create_test_remote if smart_proxy.pulp_primary? && repository.root.url
           plan_self(:repository_id => repository.id, :smart_proxy_id => smart_proxy.id)
         end
 
-        def run
+        def invoke_external_task
           repo = ::Katello::Repository.find(input[:repository_id])
-          output[:response] = repo.backend_service(smart_proxy).update_remote
+          repo.backend_service(smart_proxy).update_remote
```

Comment:
I got an error here after trying to clear out the Upstream URL for my Ansible Collections repo:

```
Action:

Actions::Pulp3::Repository::UpdateRemote

Input:

{"repository_id"=>42,
 "smart_proxy_id"=>1,
 "remote_user"=>"admin",
 "remote_cp_user"=>"admin",
 "locale"=>"en",
 "current_request_id"=>"3fddebde-3fa6-43a9-93b0-18fc163d62a5",
 "current_timezone"=>"America/New_York",
 "current_organization_id"=>1,
 "current_location_id"=>2,
 "current_user_id"=>4}

Output:

{}

Exception:

NoMethodError: super: no superclass method `remote_options' for #<Katello::Pulp3::Repository::AnsibleCollection:0x0000000016d90a10>
Did you mean?  ssl_remote_options

Backtrace:

/home/vagrant/katello/app/services/katello/pulp3/repository/ansible_collection.rb:9:in `remote_options'
/home/vagrant/katello/app/services/katello/pulp3/repository.rb:115:in `update_remote'
/home/vagrant/katello/app/lib/actions/pulp3/repository/update_remote.rb:12:in `invoke_external_task'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/action/polling.rb:84:in `initiate_external_action'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/action/polling.rb:19:in `run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/action/cancellable.rb:14:in `run'
/home/vagrant/katello/app/lib/actions/pulp3/abstract_async_task.rb:10:in `run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/action.rb:572:in `block (3 levels) in execute_run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:27:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware.rb:19:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware.rb:32:in `run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:23:in `call'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:27:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware.rb:19:in `pass'
/home/vagrant/katello/app/lib/actions/middleware/remote_action.rb:16:in `block in run'
/home/vagrant/katello/app/lib/actions/middleware/remote_action.rb:40:in `block in as_remote_user'
/home/vagrant/katello/app/models/katello/concerns/user_extensions.rb:21:in `cp_config'
/home/vagrant/katello/app/lib/actions/middleware/remote_action.rb:27:in `as_cp_user'
/home/vagrant/katello/app/lib/actions/middleware/remote_action.rb:39:in `as_remote_user'
/home/vagrant/katello/app/lib/actions/middleware/remote_action.rb:16:in `run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:23:in `call'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:27:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware.rb:19:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/foreman-tasks-5.0.0/app/lib/actions/middleware/rails_executor_wrap.rb:14:in `block in run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activesupport-6.0.3.7/lib/active_support/execution_wrapper.rb:88:in `wrap'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/foreman-tasks-5.0.0/app/lib/actions/middleware/rails_executor_wrap.rb:13:in `run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:23:in `call'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:27:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware.rb:19:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/action/progress.rb:31:in `with_progress_calculation'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/action/progress.rb:17:in `run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:23:in `call'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:27:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware.rb:19:in `pass'
/home/vagrant/katello/app/lib/actions/middleware/keep_locale.rb:11:in `block in run'
/home/vagrant/katello/app/lib/actions/middleware/keep_locale.rb:22:in `with_locale'
/home/vagrant/katello/app/lib/actions/middleware/keep_locale.rb:11:in `run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:23:in `call'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:27:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware.rb:19:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/foreman-tasks-5.0.0/app/lib/actions/middleware/load_setting_values.rb:20:in `run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:23:in `call'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:27:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware.rb:19:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/foreman-tasks-5.0.0/app/lib/actions/middleware/keep_current_request_id.rb:15:in `block in run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/foreman-tasks-5.0.0/app/lib/actions/middleware/keep_current_request_id.rb:52:in `restore_current_request_id'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/foreman-tasks-5.0.0/app/lib/actions/middleware/keep_current_request_id.rb:15:in `run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:23:in `call'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:27:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware.rb:19:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/foreman-tasks-5.0.0/app/lib/actions/middleware/keep_current_timezone.rb:15:in `block in run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/foreman-tasks-5.0.0/app/lib/actions/middleware/keep_current_timezone.rb:44:in `restore_curent_timezone'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/foreman-tasks-5.0.0/app/lib/actions/middleware/keep_current_timezone.rb:15:in `run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:23:in `call'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:27:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware.rb:19:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/foreman-tasks-5.0.0/app/lib/actions/middleware/keep_current_taxonomies.rb:15:in `block in run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/foreman-tasks-5.0.0/app/lib/actions/middleware/keep_current_taxonomies.rb:45:in `restore_current_taxonomies'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/foreman-tasks-5.0.0/app/lib/actions/middleware/keep_current_taxonomies.rb:15:in `run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:23:in `call'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:27:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware.rb:19:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware.rb:32:in `run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:23:in `call'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:27:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware.rb:19:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/foreman-tasks-5.0.0/app/lib/actions/middleware/keep_current_user.rb:15:in `block in run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/foreman-tasks-5.0.0/app/lib/actions/middleware/keep_current_user.rb:54:in `restore_curent_user'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/foreman-tasks-5.0.0/app/lib/actions/middleware/keep_current_user.rb:15:in `run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/stack.rb:23:in `call'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/middleware/world.rb:31:in `execute'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/action.rb:571:in `block (2 levels) in execute_run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/action.rb:570:in `catch'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/action.rb:570:in `block in execute_run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/action.rb:473:in `block in with_error_handling'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/action.rb:473:in `catch'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/action.rb:473:in `with_error_handling'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/action.rb:565:in `execute_run'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/action.rb:286:in `execute'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/execution_plan/steps/abstract_flow_step.rb:18:in `block (2 levels) in execute'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/execution_plan/steps/abstract.rb:167:in `with_meta_calculation'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/execution_plan/steps/abstract_flow_step.rb:17:in `block in execute'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/execution_plan/steps/abstract_flow_step.rb:32:in `open_action'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/execution_plan/steps/abstract_flow_step.rb:16:in `execute'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/director.rb:69:in `execute'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/executors/parallel/worker.rb:15:in `block in on_message'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/executors.rb:18:in `run_user_code'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/executors/parallel/worker.rb:14:in `on_message'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/context.rb:46:in `on_envelope'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/executes_context.rb:7:in `on_envelope'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/abstract.rb:25:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/actor.rb:122:in `on_envelope'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/abstract.rb:25:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/awaits.rb:15:in `on_envelope'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/abstract.rb:25:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/sets_results.rb:14:in `on_envelope'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/dynflow-1.5.0/lib/dynflow/actor.rb:56:in `on_envelope'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/abstract.rb:25:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/buffer.rb:38:in `process_envelope'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/buffer.rb:31:in `process_envelopes?'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/buffer.rb:20:in `on_envelope'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/abstract.rb:25:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/termination.rb:55:in `on_envelope'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/abstract.rb:25:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/removes_child.rb:10:in `on_envelope'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/abstract.rb:25:in `pass'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/behaviour/sets_results.rb:14:in `on_envelope'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/core.rb:162:in `process_envelope'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/core.rb:96:in `block in on_envelope'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/core.rb:119:in `block (2 levels) in schedule_execution'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-1.1.9/lib/concurrent-ruby/concurrent/synchronization/mutex_lockable_object.rb:47:in `block in synchronize'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-1.1.9/lib/concurrent-ruby/concurrent/synchronization/mutex_lockable_object.rb:47:in `synchronize'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-1.1.9/lib/concurrent-ruby/concurrent/synchronization/mutex_lockable_object.rb:47:in `synchronize'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-edge-0.6.0/lib/concurrent-ruby-edge/concurrent/actor/core.rb:116:in `block in schedule_execution'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-1.1.9/lib/concurrent-ruby/concurrent/executor/serialized_execution.rb:18:in `call'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-1.1.9/lib/concurrent-ruby/concurrent/executor/serialized_execution.rb:96:in `work'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-1.1.9/lib/concurrent-ruby/concurrent/executor/serialized_execution.rb:77:in `block in call_job'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-1.1.9/lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb:363:in `run_task'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-1.1.9/lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb:352:in `block (3 levels) in create_worker'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-1.1.9/lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb:335:in `loop'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-1.1.9/lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb:335:in `block (2 levels) in create_worker'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-1.1.9/lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb:334:in `catch'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/concurrent-ruby-1.1.9/lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb:334:in `block in create_worker'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/logging-2.3.0/lib/logging/diagnostic_context.rb:474:in `block in create_with_logging_context'
```

Reply:
good find!  I think this is an issue in master, but an easy fix!

Reply:
This should be fixed now, i will work on fixing the vcr tests failures in the meantime.